libs/knowledge-store/notebooks/support.ipynb

```diff
@@ -0,0 +1,596 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1368"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Use sitemaps to crawl the content\n",
+    "SITEMAPS = [\n",
+    "    \"https://docs.datastax.com/en/sitemap-astra-db-vector.xml\",\n",
+    "    \"https://docs.datastax.com/en/sitemap-cql.xml\",\n",
+    "    \"https://docs.datastax.com/en/sitemap-dev-app-drivers.xml\",\n",
+    "    \"https://docs.datastax.com/en/sitemap-glossary.xml\",\n",
+    "    \"https://docs.datastax.com/en/sitemap-astra-db-serverless.xml\"\n",
+    "]\n",
+    "\n",
+    "# Additional URLs to crawl for content.\n",
+    "EXTRA_URLS = [\n",
+    "    \"https://github.com/jbellis/jvector\"\n",
+    "]\n",
+    "\n",
+    "SITE_PREFIX = \"astra\"\n",
+    "\n",
+    "from bs4 import BeautifulSoup\n",
+    "import requests\n",
+    "\n",
+    "def load_pages(sitemap_url):\n",
+    "    r = requests.get(sitemap_url,\n",
+    "                     headers={\n",
+    "                         # Astra docs only return a sitemap with a user agent set.\n",
+    "                         \"User-Agent\": \"Mozilla/5.0 (X11; Linux x86_64; rv:58.0) Gecko/20100101 Firefox/58.0\",\n",
+    "                     })\n",
+    "    xml = r.text\n",
+    "\n",
+    "    soup = BeautifulSoup(xml, features=\"xml\")\n",
+    "    url_tags = soup.find_all(\"url\")\n",
+    "    for url in url_tags:\n",
+    "        yield(url.find(\"loc\").text)\n",
+    "\n",
+    "\n",
+    "# For maintenance purposes, we could check only the new articles since a given time.\n",
+    "URLS = [\n",
+    "    url\n",
+    "    for sitemap_url in SITEMAPS\n",
+    "    for url in load_pages(sitemap_url)\n",
+    "] + EXTRA_URLS\n",
+    "len(URLS)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load the content from each URL\n",
+    "Next, we create the code to load each page. This performs the following steps:\n",
+    "\n",
+    "1. Parses the HTML with BeautifulSoup\n",
+    "2. Locates the \"content\" of the HTML using an appropriate selector based on the URL\n",
+    "3. Find the link (`<a href=\"...\">`) tags in the content and collect the absolute URLs (for creating edges)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.document_loaders import AsyncHtmlLoader\n",
+    "from bs4 import BeautifulSoup\n",
+    "from langchain_core.documents import Document\n",
+    "from typing import AsyncIterator, Iterable, Set\n",
+    "from markdownify import MarkdownConverter\n",
+    "from urllib.parse import urlparse, urljoin, urldefrag\n",
+    "\n",
+    "def parse_url(link,\n",
```

Comment:
Would be good to move some of these larger code blocks into the library. At least things like "get all the `a href` links in an html body" should be pretty re-usable. I'll probably do that in a follow-up.